### PR TITLE
s3 sources: Add testdrive support for S3 and related tests

### PR DESCRIPTION
**Pull request by benesch**
This is a candidate minimum viable extraction as mentioned in #5331, since I don’t like to suggest things I haven’t tried! 

@quodlibetor this passes tests locally, and I believe it would also pass tests in CI if it had permissions to do so. I know a later change in that PR was attempting to work around some permissions issues, but I’m pretty sure they are surmountable per these docs: https://docs.aws.amazon.com/AmazonS3/latest/dev/s3-arn-format.html. It is preferable if testdrive does not assume that any resources are present so that it can be easily run on other AWS accounts.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/5460)

### CODE REVIEW

File: src/testdrive/Cargo.toml [hunk 1]
```diff
@@ -35,6 +35,7 @@ reqwest = { version = "0.11.0", features = ["native-tls-vendored"] }
 rusoto_core = { git = "https://github.com/rusoto/rusoto.git" }
 rusoto_credential = { git = "https://github.com/rusoto/rusoto.git" }
 rusoto_kinesis = { git = "https://github.com/rusoto/rusoto.git" }
+rusoto_s3 = { git = "https://github.com/rusoto/rusoto.git" }
 rusoto_sts = { git = "https://github.com/rusoto/rusoto.git" }
```
Comment (cirego) [inline src/testdrive/Cargo.toml:39]: Why are there multiple entries pointing at the same GitHub repo? And how is this different from something like `features = ["s3", "sts"]`?

**Review by cirego — COMMENTED** [review]
How many of these jobs do we plan to run in parallel? There is a 100 bucket limit in each AWS account, so that's something to (maybe) be aware of.

**Comment by quodlibetor** [comment]
I will do something like this in the future, but I'm going to close this for now to try and get everything working, and then split out reviewable chunks.

**Comment by benesch** [comment]
Reopening to debug new CI credentials.

**Pull request closed without merging**